shared/components/pages/food/food.component.js

```diff
@@ -38,26 +48,17 @@ class FoodComponent extends Panel {
   }
 
   toggleLeadersChart() {
-    let food = this;
-    food.state_manager.state.leaders_chart.show = true;
-    food.state_manager.state.leaders_chart.category = "food";
-    food.state_manager.syncLayout();
-    window.jQuery("html, body").animate({ scrollTop: $(".cc_leaders").offset().top }, 1000);
-  }
-
-  /*
-   * React Events
-   */
-
-  render(){
-    return template.call(this);
-  }
+    let food = this,
+        ui = {
+          id: 'leaders_chart',
+          data: {
+            show: true,
+            category: 'food'
+          }
+        };
 
-  componentDidMount(){
-    let food = this;
-    RELEVANT_API_KEYS.forEach((food_type)=>{
-      food.initializeSlider(food_type)
-    });
+    food.props.updateUI(ui);
+    window.jQuery('html, body').animate({ scrollTop: $('.cc_leaders').offset().top }, 1000);
```

Comment:
I moved it to the Panel component. Actually the category doesn't need to be set by each individual component because it is set by the current route in the Leaders component.